### PR TITLE
21208 - Muted microphone indicator

### PR DESCRIPTION
**Pull request by Triplelexx**
Added one script:

-  hifi/scripts/system/audioMuteOverlay.js

**Test Plan:**
Running the client script should create an overlay to provide mute feedback.

When muted a spherical overlay should be visible in the field of view of the user in first person.
It should only be visible when muted. It's should also update it's position to remain in view, as well as change color from grey to red.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by claudioshg** [comment]
Testing

**Comment by PhilipRosedale** [comment]
Testing results:  Ball/indicator moves in weird/wrong way because it's position is relative to avatar, should be relative to camera.  Try jumping or going in and out of HMD (watch the ball move down as avatar crouches in 'away' mode).

**Comment by PhilipRosedale** [comment]
QA Fail:  Ball now is in correct place, but other problems (see code review):  

1.  In interface exited in muted state and restarted, overlay is not created.  
2.  See other comments in code review.

**Comment by Triplelexx** [comment]
@PhilipRosedale I'm am unsure which additional comments you mention. I have made further changes for review.

**Comment by cozza13** [comment]
ok to test

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5883bd39dc255fcdb28e282c52fdbb664d6f6ff0 available at: https://gist.github.com/02c92d3c67ee58fe7e8cbe68a766c267<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9871-5883bd3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9871-5883bd3.dmg

**Comment by claudioshg** [comment]
Testing

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift

**Tested:** Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9871-5883bd3.exe

- PASS: When muted a spherical overlay should be visible in the field of view of the user in first person.
- PASS: It should only be visible when muted.
- FAIL:  It's should also update it's position to remain in view.
- PASS: change color from gray to red.

**Fails:**
- The sphere struggles to catch up with me as I fly at full speed. It falls behind and out of my field of view. The sphere fails to update it's position.

**Comment by PhilipRosedale** [comment]
I'm OK with the failure while flying.  Assume no flying while muted.

File: scripts/system/audioMuteOverlay.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+//
+// Created by Triplelexx on 17/03/09
+// Copyright 2017 High Fidelity, Inc.
+//
+// Distributed under the Apache License, Version 2.0.
+// See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+(function() { // BEGIN LOCAL_SCOPE
+    var utilsPath = Script.resolvePath('../developer/libraries/utils.js?v='+  Date.now());
```
Comment (ZappoMan) [inline scripts/system/audioMuteOverlay.js]: NO!!!! remove this Date stuff... unacceptable.

Comment (Triplelexx) [inline scripts/system/audioMuteOverlay.js]: I was waiting to push that out. It was left in by mistake.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fa606446a794bdfec7b7ce430df8a62f56ad681b available at: https://gist.github.com/02c92d3c67ee58fe7e8cbe68a766c267<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9871-fa60644.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9871-fa60644.dmg

**Pull request merged**